src/fromager/packagesettings.py

```diff
@@ -493,18 +493,38 @@ def plugin(self) -> types.ModuleType | None:
                 self._plugin_module = None
         return self._plugin_module
 
-    def get_patches(self) -> PatchMap:
+    def get_all_patches(self) -> PatchMap:
         """Get a mapping of version to list of patches"""
         if self._patches is None:
             patches: PatchMap = {}
-            pattern = f"{self.override_module_name}-*"
-            prefix_len = len(pattern) - 1
+            version: Version | None
+            pattern = f"{self.override_module_name}*"
```

Comment:
This pattern is going to find patches for `foo-bar` when it the module is `foo`.

Reply:
Patch directories use module override names, so `foo-bar` is normalized to `foo_bar`.

Reply:
I added a test and fix in #493